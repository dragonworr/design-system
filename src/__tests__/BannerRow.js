import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import { BannerRow, Icon, CloseButton, theme } from '..'

const bannerRow = (
  <BannerRow
    showIcon
    showCloseButton
    icon={<Icon theme={theme} />}
    text="text"
    closeButton={<CloseButton theme={theme} />}
    textAlign="left"
    theme={theme}
  />
)

describe('BannerRow', () => {
  test('renders with no props other than theme', () => {
    const tree = renderer.create(<BannerRow theme={theme} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('renders with props', () => {
    const tree = renderer.create(bannerRow).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('icon does render if showIcon is true', () => {
    const wrapper = shallow(bannerRow)
    const icon = wrapper.find('Icon')
    expect(icon).toHaveLength(1)
  })

  test('icon does not render if showIcon is false', () => {
    const wrapper = shallow(bannerRow)
    wrapper.setProps({ showIcon: false })
    const icon = wrapper.find('Icon')
    expect(icon).toHaveLength(0)
  })

  test('closeButton does render if showCloseButton is true', () => {
    const wrapper = shallow(bannerRow)
    const closeButton = wrapper.find('CloseButton')
    expect(closeButton).toHaveLength(1)
  })

  test('closeButton does not render if showCloseButton is false', () => {
    const wrapper = shallow(bannerRow)
    wrapper.setProps({ showCloseButton: false })
    const closeButton = wrapper.find('CloseButton')
    expect(closeButton).toHaveLength(0)
  })
})
