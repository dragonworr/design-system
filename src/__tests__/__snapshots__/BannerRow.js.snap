// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerRow renders with no props other than theme 1`] = `
.c1 {
  text-align: center;
}

.c2 {
  width: 100%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
    width={1}
  />
</div>
`;

exports[`BannerRow renders with props 1`] = `
.c1 {
  text-align: center;
}

.c2 {
  width: 100%;
  text-align: left;
}

.c4 {
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  text-decoration: none;
  font-weight: 600;
  cursor: pointer;
  border-radius: 2px;
  background-color: #007aff;
  color: #fff;
  border-width: 0;
  border-style: solid;
  height: 40px;
  font-size: 14px;
  padding: 0 18px;
}

.c4:disabled {
  opacity: 0.25;
}

.c4:hover {
  background-color: #049;
}

.c3 {
  padding: 0;
  height: auto;
  background-color: transparent;
  color: currentcolor;
}

.c3 hover: {
  background-color: transparent;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div
  className="c0 c1"
>
  <svg
    className=""
    fill="currentcolor"
    height={24}
    theme={
      Object {
        "bold": 700,
        "boxShadows": Array [
          "0 0 4px 0 rgba(0,0,0,0.08)",
          "0 8px 8px 0 rgba(0,0,0,0.08)",
          "0 16px 16px 0 rgba(0,0,0,0.08)",
          "0 32px 32px 0 rgba(0,0,0,0.08)",
        ],
        "breakpoints": Array [
          32,
          40,
          48,
          64,
        ],
        "colors": Object {
          "black": "#000",
          "blue": "#007aff",
          "blue0": "#cdf",
          "blue1": "#cdf",
          "blue2": "#007aff",
          "blue3": "#007aff",
          "blues": Array [
            "#cdf",
            "#cdf",
            "#007aff",
            "#007aff",
          ],
          "borderGray": "#abb7c2",
          "darkBlue": "#049",
          "darkGray": "#364049",
          "darkGreen": "#060",
          "darkOrange": "#950",
          "darkPurple": "#407",
          "darkRed": "#800",
          "gray": "#687B8E",
          "gray0": "#f0f2f4",
          "gray1": "#f0f2f4",
          "gray2": "#687B8E",
          "gray3": "#687B8E",
          "green": "#0a0",
          "green0": "#cec",
          "green1": "#cec",
          "green2": "#0a0",
          "green3": "#0a0",
          "greens": Array [
            "#cec",
            "#cec",
            "#0a0",
            "#0a0",
          ],
          "lightBlue": "#cdf",
          "lightGray": "#f0f2f4",
          "lightGreen": "#cec",
          "lightOrange": "#fec",
          "lightPurple": "#ecf",
          "lightRed": "#fcc",
          "orange": "#f90",
          "orange0": "#fec",
          "orange1": "#fec",
          "orange2": "#f90",
          "orange3": "#f90",
          "oranges": Array [
            "#fec",
            "#fec",
            "#f90",
            "#f90",
          ],
          "purple": "#70b",
          "purple0": "#ecf",
          "purple1": "#ecf",
          "purple2": "#70b",
          "purple3": "#70b",
          "purples": Array [
            "#ecf",
            "#ecf",
            "#70b",
            "#70b",
          ],
          "red": "#c00",
          "red0": "#fcc",
          "red1": "#fcc",
          "red2": "#c00",
          "red3": "#c00",
          "reds": Array [
            "#fcc",
            "#fcc",
            "#c00",
            "#c00",
          ],
          "text": "#001833",
          "white": "#fff",
        },
        "font": "'Helvetica Neue',Helvetica,Arial,sans-serif",
        "fontSizes": Array [
          12,
          14,
          16,
          20,
          24,
          32,
          48,
        ],
        "maxContainerWidth": "1280px",
        "mediaQueries": Array [
          "@media screen and (min-width:32em)",
          "@media screen and (min-width:40em)",
          "@media screen and (min-width:48em)",
          "@media screen and (min-width:64em)",
        ],
        "radius": "2px",
        "regular": 400,
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
        ],
      }
    }
    viewBox="0 0 20 19"
    width={24}
  >
    <path
      d="M19.999 1.42a1.374 1.374 0 0 0-.572-1.117 1.468 1.468 0 0 0-.975-.3c-.386.025-.749.21-.99.51L6.854 15.471l-4.422-4.476c-.556-.543-1.505-.534-2.037.024A1.457 1.457 0 0 0 0 12.055c.008.397.161.77.42 1.028l5.476 5.512c.266.26.637.405 1.031.405A1.353 1.353 0 0 0 8 18.466L19.62 2.439c.25-.26.387-.631.379-1.02h-.001z"
    />
  </svg>
  <div
    className="c2"
    width={1}
  >
    text
  </div>
  <button
    className="c3 c4"
    name="closeLight"
    onClick={undefined}
    size={14}
    title="close"
  >
    <svg
      className=""
      color={undefined}
      fill="currentcolor"
      height={14}
      viewBox="0 0 20 20"
      width={14}
    >
      <path
        d="M18.127 19.817c-.33 0-.66-.127-.9-.378L1.031 2.514A1.116 1.116 0 0 1 1.113.882 1.262 1.262 0 0 1 2.83.96l16.196 16.925a1.116 1.116 0 0 1-.082 1.632 1.25 1.25 0 0 1-.818.301v-.001z M1.958 19.747c-.295 0-.59-.101-.824-.306a1.115 1.115 0 0 1-.07-1.632L17.391.997a1.262 1.262 0 0 1 1.717-.066c.494.432.525 1.163.07 1.632L2.852 19.375c-.24.247-.567.372-.894.372z"
      />
    </svg>
  </button>
</div>
`;
